Document/0x06b-Basic-Security-Testing.md

```diff
@@ -984,8 +990,38 @@ $ rvictl -s <UDID>
+Next, launch Wireshark and select "rvi0" as the capture interface.
+
+#### Network Traffic Capture with SSH on a Jailbroken Device
+
+You can remotely sniff traffic in real-time on iOS with [tcpdump and Wireshark](https://blog.jjhayes.net/wp/2019/02/28/capture-iphone-network-traffic-with-tcpdump-and-wireshark/) over the SSH protocol. This method requires a jailbroken iOS device, however it can be performed from any operating system. First, make sure you have [Wireshark](0x08a-Testing-Tools.md#wireshark) installed on your host computer.
+
+1. Open Cydia and install the `tcpdump` and `OpenSSH` packages.
+1. Connect your iOS device and your computer to the same network.
+1. Install and open Wireshark on your host computer.
+1. Select the cog icon next "SSH remote capture: sshdump" at the initial "Capture" UI page.
+1. Within the "Server" tab:
+    1. Enter your iOS device's IP address as "Remote SSH server address".
+    1. Enter `22` as "Remote SSH server port".
+1. Within the "Authentication" tab:
+    1. Enter `root` as the "Remote SSH server username".
+    1. Enter the "SSH server password" (see: [Remote Shell](#remote-shell)).
+1. Within the "Capture" tab:
+    1. Ensure that "Remote capture command selection" is `tcpdump`.
+    2. Optionally, set "Remote capture filter" to `not port 22` to eliminate noise from Wireshark's SSH connection to the iOS device.
+1. Select "Save", double click "SSH remote capture: sshdump" and perform your application testing.
+
+After testing is complete, click the red "Stop" button and use "File" > "Save As" to retain the capture data as a `PCAP` file for later analysis.
+
+#### Decrypting Captured Packets
+
+HTTPS traffic captured in the PCAP can be decrypted for review within Wireshark if testing was performed with an HTTP proxy supporting `SSLKEYLOGFILE` key logging. This is possible with [mitmproxy](https://docs.mitmproxy.org/stable/howto-wireshark-tls/), but not Burp Suite ([request](https://forum.portswigger.net/thread/option-to-create-nss-key-log-file-d57f526f)) or ZAP ([issue](https://github.com/zaproxy/zaproxy/issues/1630)). Navigate to Wireshark's `Preferences > Protocols > TLS` configuration and use the "(Pre)-Master-Secret log filename" field to browse to the `SSLKEYLOGFILE` file prepared by your HTTP proxy.
+
+Alternatively to generating `SSLKEYLOGFILE` from web proxy, the Frida script ["ios-tls-keylogger.js"](https://codeshare.frida.re/@andydavies/ios-tls-keylogger/) can sometimes be used to generate this file when iOS apps use common SSL/TLS libraries. This is considered a more fragile approach as the `CALLBACK_OFFSET` constant must be [updated in the script](https://www.mustafadur.com/blog/intercepting-ssl-and-https/#sniffing-with-sslkeylogfile) to match your tested iOS version, and not all client-side implementations of SSL/TLS will work.
```

Comment:
```suggestion
As an alternative to generating `SSLKEYLOGFILE` from the web proxy, the Frida script ["ios-tls-keylogger.js"](https://codeshare.frida.re/@andydavies/ios-tls-keylogger/) can sometimes be used to generate this file when iOS apps use common SSL/TLS libraries. This is considered a more fragile approach as the `CALLBACK_OFFSET` constant must be [updated in the script](https://www.mustafadur.com/blog/intercepting-ssl-and-https/#sniffing-with-sslkeylogfile) to match your tested iOS version, and not all client-side implementations of SSL/TLS will work.
```
Additionally, this approach works, but when/why would you need to do this? What does this ("fragile") approach offer that others don't?